data-plane/src/config_client.rs

```diff
@@ -122,6 +123,50 @@ impl ConfigClient {
         }
     }
 
+    pub async fn get_object(&self, key: String) -> Result<GetObjectResponse> {
+        let payload = GetObjectRequest::new(key).into_body()?;
+
+        let response = self.send(ConfigServerPath::Storage, "GET", payload).await?;
+
+        let result: GetObjectResponse = self.parse_response(response).await?;
+
+        Ok(result)
+    }
+
+    pub async fn put_object(&self, key: String, object: String) -> Result<()> {
+        let payload = PutObjectRequest::new(key, object).into_body()?;
+
+        let response = self.send(ConfigServerPath::Storage, "PUT", payload).await?;
+
+        if response.status() == StatusCode::OK {
+            Ok(())
+        } else {
+            println!("Error in message send to control plane");
```

Comment:
Would be good to add status code to the errors here

Reply:
Good shout, updated